include/unifex/linux/io_epoll_context.hpp

```diff
@@ -0,0 +1,520 @@
+  void schedule_remote(operation_base* op) noexcept;
+
+  // Insert the timer operation into the queue of timers.
+  // Must be called from the I/O thread.
+  void schedule_at_impl(schedule_at_operation* op) noexcept;
+
+  // Execute all ready-to-run items on the local queue.
+  // Will not run other items that were enqueued during the execution of the
+  // items that were already enqueued.
+  // This bounds the amount of work to a finite amount.
+  void execute_pending_local() noexcept;
+
+  // Check if any completion queue items are available and if so add them
+  // to the local queue.
+  void acquire_completion_queue_items() noexcept;
+
+  // Check if any completion queue items have been enqueued and move them
+  // to the local queue.
+  void acquire_remote_queued_items() noexcept;
+
+  // Submit a request to the submission queue containing an IORING_OP_POLL_ADD
+  // for the remote queue eventfd as a way of registering for asynchronous
+  // notification of someone enqueueing
+  //
+  // Returns true if successful. If so then it is no longer permitted
+  // to call 'acquire_remote_queued_items()' until after the completion
+  // for this POLL_ADD operation is received.
+  //
+  // Returns false if either no more operations can be submitted at this
+  // time (submission queue full or too many pending completions) or if
+  // some other thread concurrently enqueued work to the remote queue.
+  bool try_register_remote_queue_notification() noexcept;
+
+  // Signal the remote queue eventfd.
+  //
+  // This should only be called after trying to enqueue() work
+  // to the remoteQueue and being told that the I/O thread is
+  // inactive.
+  void signal_remote_queue();
+
+  void remove_timer(schedule_at_operation* op) noexcept;
+  void update_timers() noexcept;
+  bool try_submit_timer_io(const time_point& dueTime) noexcept;
+  bool try_submit_timer_io_cancel() noexcept;
+
+  std::uintptr_t timer_user_data() const {
+    return reinterpret_cast<std::uintptr_t>(&timers_);
+  }
+
+  std::uintptr_t remove_timer_user_data() const {
+    return reinterpret_cast<std::uintptr_t>(&currentDueTime_);
+  }
+
+  ////////
+  // Data that does not change once initialised.
+
+  // Resources
+  safe_file_descriptor epollFd_;
+  safe_file_descriptor timerFd_;
+  safe_file_descriptor remoteQueueEventFd_;
+
+  ///////////////////
+  // Data that is modified by I/O thread
+
+  // Local queue for operations that are ready to execute.
+  operation_queue localQueue_;
+
+  // Set of operations waiting to be executed at a specific time.
+  timer_heap timers_;
+
+  // The time that the current timer operation submitted to the kernel
+  // is due to elapse.
+  std::optional<time_point> currentDueTime_;
+
+  bool remoteQueueReadSubmitted_ = false;
+  bool timersAreDirty_ = false;
+
+  std::uint32_t activeTimerCount_ = 0;
+
+  static const std::uint32_t maxCount_ = 256;
+  epoll_event completed_[maxCount_];
+  std::uint32_t count_;
```

Comment:
Can we remove these data-members and just make them local variables inside `acquire_completion_queue_items()`?